### DISCUSSION TITLE
Proper Dialog example?

### DISCUSSION THREAD

**Discussion by M0n7y5** [HandyOrg/HandyControl#1173] [Q&A] [answered]
I cant find any proper example for using dialogs.  The one in docs is incomplete. It doesnt explain how provided viewmodel should look like and it works somw DialogResult variable that is not defined anywhere. Can somebody explain to me how to properly await on dialog result? 

![obrazek](https://user-images.githubusercontent.com/17201053/165975610-5ebdcb71-0909-447b-9c60-ddff59f3d84a.png)

I am using Interactions from ReactiveUI and my current code looks like this

![obrazek](https://user-images.githubusercontent.com/17201053/165976210-d06b694b-1a3b-4fc7-aed4-b2249aadc7bc.png)

However this gives me exception: 
![obrazek](https://user-images.githubusercontent.com/17201053/165976397-9026993d-2ae3-407f-aa10-b2a0612fc507.png)

**Comment by Racer396** [comment, marked as answer]
I don't know, is question still actual, but I got the same error, tried to Google and found only this.
After that, I started researching the source code of the demo.

**Simple answer**: just set DataContext in your dialog view (`UserControl`) `DataContext = YourViewModel` (VM should inherit `IDialogResultable`/`IDialogResultable<T>`)

**Details**: In the demo (I get `InteractiveDialog` as an example) they do this by `DataContext="{Binding InteractiveDialog, Source={StaticResource Locator}}"`, but `IntercativeDialog` is here is not view by itself, this is the property in `ViewModelLocator`, that looks like this:
`public InteractiveDialogViewModel InteractiveDialog => SimpleIoc.Default.GetInstance<InteractiveDialogViewModel>();`

**Reply by jotalanusse** [reply]
Hi there,

I must admit that I'm still struggling to understand how to implement this solution in my project.

I would greatly appreciate it if you could provide additional explanations or perhaps even an example of how to set the DataContext property to the ViewModel that implements the IDialogResultable or IDialogResultable<T> interface.

Thank you in advance for your help.
